src/main/scala/edu/ie3/mobsim/MobilitySimulator.scala

```diff
@@ -615,7 +622,7 @@ object MobilitySimulator
     * @return
     *   Time tick (as ArrayList) when simulation should be triggered again
     */
-  protected def doActivity(tick: Long): java.util.ArrayList[java.lang.Long] = {
+  protected def doActivity(tick: Long): java.util.ArrayList[Long] = {
```

Comment:
I don't think this works, as the implementation here has to follow the exact same method signature. So this is a point where we still need `java.lang.Long`

Reply:
In addition to the mentioned line, there are some more lines where the type needs to be ``java.lang.Long`` because the return value needs to match the code in the simona api.